pkg/promtail/targets/file/filetarget.go

```diff
@@ -316,6 +317,9 @@ func (t *FileTarget) stopTailingAndRemovePosition(ps []string) {
 			t.positions.Remove(tailer.path)
 			delete(t.tails, p)
 		}
+		if h, ok := t.handler.(api.InstrumentedEntryHandler); ok {
+			h.UnregisterLatencyMetric(model.LabelSet{model.LabelName(client.LatencyLabel): model.LabelValue(p)})
```

Comment:
I don't see the host label here, meaning it won't delete anything because the fingerprint won't match.

cyrilgate lol

Reply:
lol, good catch, i added the host label and haven't tested since adding it